plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -3645,13 +3834,31 @@ private OptionalLong executeDelete(ConnectorSession session, ConnectorTableHandl
                     tableHandle.getReadVersion(),
                     tableHandle.getMetadataEntry().getCheckpointInterval(),
                     commitVersion);
+            cacheTableMetadataAsync(tableHandle.getSchemaName(), tableHandle.getTableName(), Optional.ofNullable(tableHandle.getMetadataEntry().getDescription()), commitVersion, tableHandle.getMetadataEntry().getSchemaString());
```

Comment:
We should do all these operations involving transactions in `TestDeltaLakeMetastoreAccessOperations` to ensure that the `REPLACE_TABLE` operation is executed.